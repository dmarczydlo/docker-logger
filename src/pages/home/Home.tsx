import React from 'react';
import clsx from 'clsx';
import Grid from '@material-ui/core/Grid';
import Paper from '@material-ui/core/Paper';
import Server from '../../components/server';
import { IServer } from '../../interfaces/docker';
import useStyles from './styles';

interface Props {
  servers?: IServer[]
}

const Home: React.SFC<Props> = (props: Props) => {

  const classes = useStyles();
  const { servers } = props;
  const fixedHeightPaper = clsx(classes.wrapper, classes.fixedHeight);

  return (
    <Grid container spacing={3}>
      {/* Chart */}
      <Grid item xs={12} md={8} lg={9}>
        <div className={classes.wrapper}>
          {servers && servers.map((server, index) => <Server key={`${server.name}${index}`} server={server} index={index} />)}
        </div>
      </Grid>
      {/* Recent Deposits */}
      <Grid item xs={12} md={4} lg={3}>
        <Paper className={fixedHeightPaper}>
          <div>PLACE FOR THIS 2</div>
        </Paper>
      </Grid>
    </Grid>
  );
}

export default Home;