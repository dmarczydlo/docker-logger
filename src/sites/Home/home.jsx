import React, {Fragment} from 'react';
import {Row, Col} from 'react-grid-system';
import Info from '../../components/info';

const displayMessageOrInfo = (message, servers) => {
    let ret = null;
    const errors = [];
    if (servers) {
        ret = (
            <Fragment>
                <Row xs={16}>
                    {servers.map((element, i) => {
                        element.error ? ( // eslint-disable-line no-unused-expressions
                            errors.push(element.error)
                        ) : null;
                        return (
                            <Fragment key={i}>
                                {element.name ? (
                                    <Col lg={3} xs={12}>
                                        <Info {...element} />
                                    </Col>
                                ) : <span />
                                }
                            </Fragment>
                        );
                    })}
                </Row>
            </Fragment>
        );
    }
    return ret;
};

const Home = (props) => {
    const {message, servers} = props;
    return (
        <div>
            <div className="homeBlock">
                <h1>Dockers logger</h1>
                <p>Open source application to display information about dockers status. Multi servers are
                    available.
                </p>
                <p>To add own servers you have to edit <b>config.json</b> file and insert own data.
                </p>
            </div>
            <div className="homeBlock">
                <h1>Available servers</h1>
                {displayMessageOrInfo(message, servers)}
            </div>
        </div>
    );
};

export default Home;
