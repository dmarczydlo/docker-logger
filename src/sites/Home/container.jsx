import {connect} from 'react-redux';
import Home from './home';

const mapStateToProps = (state) => {
    const {dockers} = state;
    return dockers;
};

const mapDispatchToProps = () => {
    return {};
};

export default connect(mapStateToProps, mapDispatchToProps)(Home);
