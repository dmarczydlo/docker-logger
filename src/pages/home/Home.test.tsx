import React from 'react';
import renderer from 'react-test-renderer';
import Home from './Home';

it('renders without crashing', () => {
  const component = renderer.create(
    <Home />,
);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
