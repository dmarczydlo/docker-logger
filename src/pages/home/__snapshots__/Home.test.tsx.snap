// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8 MuiGrid-grid-lg-9"
  >
    <div
      className="makeStyles-wrapper-1"
    />
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 MuiGrid-grid-lg-3"
  >
    <div
      className="MuiPaper-root MuiPaper-elevation1 makeStyles-wrapper-1 makeStyles-fixedHeight-2 MuiPaper-rounded"
    >
      <div>
        PLACE FOR THIS 2
      </div>
    </div>
  </div>
</div>
`;
